module-2/backend/week_one.md

```diff
@@ -1,32 +1,146 @@
+
+Why do I need a development AND test database?
+* The test database is for test and it is destroyed right after where as a development database is for non-testing use so it is not destroyed unless given a command
+
+What's responsive design?
+* Ability for your design to be sized properly on different devices with our sacrificing the functionality. It will properly fit if you adjust the size of your browser.
+
+What is CRUD and why is it important?
+* C-create R-read U-update D-delete. These give us our basic actions we can do to our database
+
+What does HTTP stand for?
+* H-hyper T-text T-i forget P-i forget
+
+What are the two ways to interpolate Ruby in an ERB view template?
+* <%= @instance_varibale %> or <% @instance_varibale %>
+
+What's the difference between these two ways?
+* If you use an = sign the ruby code will appear to the client but if you do not then that does not appear to the client
+
+ What's an ORM?
+ * Object relational  mapping
+
+What's the most commonly used ORM?
+* Active record
+
+Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+```
+get '/' do
+    erb: dashboard
   end
   ```
+  This will be your welcome page which desplays no info from the db
+
+* One
+ ```
+get '/restaurants' do
+      @restaurants = Restaurants.all
+      erb: restaurants.index
+    end
+```
+
+This shows us all the restaurants that were created
+
+* two
+```
+get '/restaurants/new' do
+    erb: new.index
+  end
+```
+
+* three
+```
+post 'restaurants/' do
+      name = Restaurant.new(params[:restaurant])
+      name.save
+      redirect '/restaurants'
+    end
+```
+
+This is when you our on restaurants/new and his the submit button in the form it created a new instance of Restaurant and saved it to the db. Your new instance of restaurant will now be displayed on the restaurant.index erb file.
+
+* four
+```
+get 'restaurant/id' do
+  @restaurant = Restaurant.find(params[:id])
+  erb :show
+  end
+```
+
+* five
+```
+get 'restaurant/:id/edit' do
+  @restaurant = Restaurant.find(params[:id])
+  erb :edit
+  end
+```
+This brings us to the edit page that shows a particular form to update the instance of Restaurant
+
+* six
+```
+put 'restaurant/id' do
+  Restaurant.update(params[:id])
+  redirect '/restaurants'
+  end
+```  
+This enables you to update a current instance and update it in the db. It also will bring you back to restaurants
+
+* seven
+```
+delete 'restaurant/id' do
+  Restaurant.destroy(params[:id])
+  redirect '/restaurants'
+  end
+```   
+This enables you to destroy an instance of restaurant
+This brings you to the show erb which only displaces that particular id or that once instance of Restaurant
+
+What's a migration?
+* migrates to our .schema. rake db:migrate

```

Comment:
A migration is the file that we create to store the instructions on how to create/update the database, but creating the migration (the file with the instructions) is a separate step from *running* the migration (which happens when you run `rake db:migrate`). It seems like you're aware of this since you included the command above, but just want to make sure that it takes **both** steps (creating and running) to update the DB.